src/Limiters/limiter.jl

```diff
@@ -0,0 +1,116 @@
+"""
+    quasimonotone_limiter!(ρstar, Qstar, minQ, maxQ, space, tol)
+
+This limiter is inspired by Guba et al Optimization-based limiters for the
+spectral element method", JCP 2014, doi: https://doi.org/10.1016/j.jcp.2014.02.029
+In the reference paper, it is denoted by OP1, and is outlined in eq.s (37)-(40).
+As in HOMME, the implementation idea here is the following: We need to find a grid field which is closest
+to the initial field (in terms of weighted sum), but satisfies the min/max constraints.
+So, first we find values which do not satisfy constraints and bring these values
+to a closest constraint. This way we introduce some mass change (`mass_change`),
+so, we redistribute `mass_change` in the way that the l2 error is smallest.
+This redistribution might violate constraints thus, we do a few iterations.
+"""
+function quasimonotone_limiter!(
+    ρstar::Fields.Field,
+    Qstar::Fields.Field,
+    space::Spaces.SpectralElementSpace2D,
+    FT,
+    minQ,
+    maxQ,
+    tol,
+)
```

Comment:
Would be good to add a unit test that just evaluates this function (this could even be on a single element)

Reply:
Done! It seems like it's doing what it's supposed to do